src/sentry/spans/span.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+import hashlib
+from dataclasses import dataclass
+from typing import Mapping, Optional, Sequence, TypedDict
+
+from sentry import nodestore
+
+
+class SpanData(TypedDict):
+    id: str
+    contexts: Mapping[str, object]
+    project_id: int
+    tags: Mapping[str, str]
+
+
+@dataclass(frozen=True)
+class Span:
+    contexts: Mapping[str, object]
+    id: str
+    project_id: int
+    tags: Mapping[str, str]
+
+    def to_dict(self) -> SpanData:
+        return {
+            "contexts": self.contexts,
+            "id": self.id,
+            "project_id": self.project_id,
+            "tags": self.tags,
+        }
+
+    @classmethod
+    def from_dict(cls, data: SpanData) -> Span:
+        return cls(
+            data["contexts"],
+            data["id"],
+            data["project_id"],
+            data["tags"],
+        )
+
+    @classmethod
+    def build_storage_identifier(cls, span_id: str, project_id: int) -> str:
+        identifier = hashlib.md5(f"{span_id}::{project_id}".encode()).hexdigest()
```

Comment:
Sort of related to Filippo's comment. But I was thinking that the key could be hierarchical something like the following.
`s:<project_id>:<trace_id>:<span_id>`

The reasons for this is following:
1. `project_id` - There are requirements to be able to delete data from all forms of storages. If we prefix `project_id` in all spans data in nodestore, it makes it easy to delete data. You can perform a prefix lookup and do this deletion easily. Today, deletions are performed by performing lookups for projects in clickhouse, retrieving all required fields and forming the individual keys. The deleting the individual keys from nodestore. We can drastically improve this case.
2. `trace_id` coming after projects makes it easier to retrieve all data corresponding to a trace more effectively. One of the use cases nodestore needs to solve for is building the full trace view. Doing the lookup by just the project and trace prefix without having to provide all span ids would make things faster.
3. `span_id` being the last since it is the leaf node.